aptos-move/framework/src/natives/dispatchable_fungible_asset.rs

```diff
@@ -0,0 +1,57 @@
+use super::function_info::extract_function_info;
+use aptos_gas_algebra::InternalGas;
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+use aptos_native_interface::{
+    RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeError, SafeNativeResult,
+};
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{loaded_data::runtime_types::Type, values::Value};
+use smallvec::SmallVec;
+use std::collections::VecDeque;
+
+/***************************************************************************************************
+ * native fun dispatchable_withdraw / dispatchable_deposit / dispatchable_derived_balance
+ *
+ *   Directs control flow based on the last argument. We use the same native function implementation
+ *   for all dispatching native.
+ *   gas cost: a flat fee because we charged the loading of those modules previously.
+ *
+ **************************************************************************************************/
+pub(crate) fn native_dispatch(
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut arguments: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    let (module_name, func_name) = extract_function_info(&mut arguments)?;
```

Comment:
I don't see the logic that actually invokes the loaded function. Isn't it supposed to?